bin/inference/pycbc_inference

```diff
@@ -293,10 +312,14 @@ with ctx:
             logging.info("Writing results to file")
             sampler.write_results(fp, start_iteration=start,
                                   end_iteration=end,
-                                  max_iterations=opts.niterations + n_burnin)
+                                  max_iterations=opts.niterations+
+                                                 n_sampler_burnin)
             logging.info("Writing state to file")
             fp.write_random_state()
 
+            logging.info("Updating burn in")
+            burnin_eval.update(sampler, fp)
```

Comment:
Why is the burn in in the run sampler loop? I don't see anything that changes ``intervals`` (eg. ``intervals += n_sampler_burnin`` in the code block above for ``use_sampler``).

What's the idea here? Is it if you say run for 10,000 iterations it will constantly recalculate the burn in but not add any extra iterations to ensure you have 10,000 samples after burn in? Is this why ``use_sampler`` is treated differently?